### PR TITLE
Add metrics onto proxy level. Add counter for Loglist updates

### PR DESCRIPTION
**Pull request by Mercurrent**

### CODE REVIEW

File: submission/proxy.go [hunk 1]
```diff
@@ -39,6 +39,16 @@ const (
 	AppleCTPolicy
 )
 
+var (
+	proxyOnce    sync.Once
+	llUpdCounter monitoring.Counter
```
Comment (phad) [inline submission/proxy.go]: ll ? 11 ?  what's this?

Comment (phad) [reply submission/proxy.go]: maybe logListUpdates  and drop the word 'Counter'.  It's explicit where you create it that it's a Counter, and later logListUpdates.Inc() makes implicit that it's a Counter too.

File: submission/proxy_server.go [hunk 2]
```diff
@@ -35,9 +36,9 @@ type ProxyServer struct {
 }
 
 // NewProxyServer creates ProxyServer instance. Call Run() to init.
-func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout time.Duration) *ProxyServer {
+func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout time.Duration, mf monitoring.MetricFactory) *ProxyServer {
 	s := &ProxyServer{addTimeout: reqTimeout}
-	s.p = NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder)
+	s.p = NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder, mf)
```
Comment (phad) [inline submission/proxy_server.go]: not in this diff (which is fine), but did you think about just writing:

```go
return &ProxyServer{
  addTimeout: reqTimeout,
  p: NewProxy(NewLogListManager(NewLogListRefresher(logListPath)), dBuilder, mf)
}
```

File: submission/proxy_test.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ func stubLogListManager() *LogListManager {
 }
 
 func TestProxyRefreshLLErr(t *testing.T) {
-	p := NewProxy(stubLogListManager(), GetDistributorBuilder(ChromeCTPolicy, NewStubLogClient, monitoring.InertMetricFactory{}))
+	p := NewProxy(stubLogListManager(), GetDistributorBuilder(ChromeCTPolicy, NewStubLogClient, monitoring.InertMetricFactory{}), monitoring.InertMetricFactory{})
```
Comment (phad) [inline submission/proxy_test.go]: Can these two inert metric factory instances be identical?
i.e., outside this test func, declare:

`var imf monitoring.InertMetricFactory`

and just pass `imf` in each of these tests wherever you currently initialise and pass an empty `InertMetricFactory`.

File: submission/proxy_server.go (hunk 2 above)
Comment (Mercurrent) [reply submission/proxy_server.go]: When you say 'not in this diff' does it mean you'd prefer a separate PR resolving it?

File: submission/proxy_test.go (hunk 3 above)
Comment (Mercurrent) [reply submission/proxy_test.go]: Done

File: submission/proxy.go (hunk 1 above)
Comment (Mercurrent) [reply submission/proxy.go]: Renamed per suggestion.

**Comment by Mercurrent** [comment]
PTAL

File: submission/proxy_server.go (hunk 2 above)
Comment (phad) [reply submission/proxy_server.go]: No, I just meant I was commenting on something you didn't introduce in this PR. No need to make a separate PR if you want to fix it now.

Comment (phad) [reply submission/proxy_server.go]: And, no need to fix if you think it's just fine as it is.

**Pull request merged**